reconcile/test/test_queris.py

```diff
@@ -0,0 +1,82 @@
+from unittest import TestCase
+from unittest.mock import create_autospec, patch
+
+from reconcile import queries
+from reconcile.utils import gql
+
+from .fixtures import Fixtures
+
+PERMISSIONS_QUERY = """
+{
+  permissions: permissions_v1 {
+    service
+    ...on PermissionSlackUsergroup_v1 {
+      channels
+      description
+      handle
+      ownersFromRepos
+      pagerduty {
+          name
+          instance {
+            name
+          }
+          scheduleID
+          escalationPolicyID
+        }
+      roles {
+        users {
+            name
+            org_username
+            slack_username
+            pagerduty_username
+        }
+    }
+      schedule {
+          schedule {
+            start
+            end
+            users {
+              org_username
+              slack_username
+            }
+          }
+        }
+      workspace {
+        name
+        token {
+          path
+          field
+        }
+        api_client {
+          global {
+            max_retries
+            timeout
+          }
+          methods {
+            name
+            args
+          }
+        }
+        managedUsergroups
+      }
+    }
+  }
+}
+"""
+
+
+class TestQueris(TestCase):
+    @patch.object(gql, "get_api")
+    def test_get_permissions_return_all_slack_usergroup(self, get_api):
+        gqlapi_mock = create_autospec(gql.GqlApi)
+        gql.get_api.return_value = gqlapi_mock
+        gqlapi_mock.query(PERMISSIONS_QUERY).return_value = \
```

Comment:
I don't think this is doing what you expect it to do. Consider this example:

```
>>> from unittest.mock import MagicMock
>>> mock = MagicMock()

# What you're doing currently
>>> mock.some_method('test').return_value = 'some-value'
>>> mock.some_method()
<MagicMock name='mock.some_method()' id='139985539334496'>
>>> mock.some_method('test')
<MagicMock name='mock.some_method()' id='139985539334496'>

# Without any arg, you're actually setting the return value of the method
# (but it doesn't care about args)
>>> mock.some_method.return_value = 'some-value'
>>> mock.some_method('test')
'some-value'
>>> mock.some_method()
'some-value'
```

If you're trying to return a certain value based on a given input, you want `side_effect` instead. I'm a bit surprised that this would be working at all, so it might be worth taking a closer look (or telling me why I'm wrong).